src/lib/tls/tls_channel.cpp

```diff
@@ -168,6 +180,15 @@ Handshake_State& Channel::create_handshake_state(Protocol_Version version)
          m_sequence_numbers.reset(new Stream_Sequence_Numbers);
       }
 
+   for (uint64_t seq = 0; seq < pre_in_message_seq; seq++)
```

Comment:
Add sanity check that if `pre_in_message_seq > 0 || pre_out_message_seq > 0` then we are in DTLS state.